(window.webpackJsonp=window.webpackJsonp||[]).push([[5],{210:function(e,t,o){e.exports=o.p+"assets/img/TRiCoLOR.1b8be74c.png"},218:function(e,t,o){"use strict";o.r(t);var i=o(28),a=Object(i.a)({},(function(){var e=this,t=e.$createElement,i=e._self._c||t;return i("ContentSlotsDistributor",{attrs:{"slot-key":e.$parent.slotKey}},[i("p",[i("img",{attrs:{src:o(210),alt:"TRiCoLOR logo"}})]),e._v(" "),i("p",[i("a",{attrs:{href:"https://github.com/davidebolo1993/TRiCoLOR",target:"_blank",rel:"noopener noreferrer"}},[e._v("TRiCoLOR"),i("OutboundLink")],1),e._v(" is an efficient command-line application capable to accurately profile tandem repeats in error-prone long-read sequencing data.\nTRiCoLOR works on data both from Oxford Nanopore Technologies and Pacific Biosciences sequencers and is fast enough to be used on whole-genome alignments.\nTRiCoLOR offers:")]),e._v(" "),i("ol",[i("li",[e._v("A module suitable to identify repetitive regions de novo (SENSoR)")]),e._v(" "),i("li",[e._v("A module suitable to profile tandem repeats (REFER)")]),e._v(" "),i("li",[e._v("A module suitable to visualize tandem repeats (ApP)")]),e._v(" "),i("li",[e._v("A module suitable to check the mendelian consistency of tandem repeats in trio genomes sequencing data (SAGE)")])]),e._v(" "),i("p",[e._v("All code is open-source and hosted on "),i("a",{attrs:{href:"https://github.com/davidebolo1993/TRiCoLOR",target:"_blank",rel:"noopener noreferrer"}},[e._v("TRiCoLOR's GitHub page"),i("OutboundLink")],1),e._v(".")]),e._v(" "),i("p",[e._v("Contents:")]),e._v(" "),i("ol",[i("li",[i("RouterLink",{attrs:{to:"/installation/installation.html"}},[e._v("Installation")])],1),e._v(" "),i("li",[i("RouterLink",{attrs:{to:"/usage/usage.html"}},[e._v("General usage")])],1),e._v(" "),i("li",[i("RouterLink",{attrs:{to:"/usecases/usecases.html"}},[e._v("Use cases")])],1)]),e._v(" "),i("div",{staticClass:"custom-block tip"},[i("p",{staticClass:"custom-block-title"},[e._v("TIP")]),e._v(" "),i("p",[e._v("For questions, help or feature requests, please open an "),i("a",{attrs:{href:"https://github.com/davidebolo1993/TRiCoLOR/issues",target:"_blank",rel:"noopener noreferrer"}},[e._v("issue"),i("OutboundLink")],1),e._v(" or get in touch with me at davidebolognini7@gmail.com")])]),e._v(" "),i("h2",{attrs:{id:"citation"}},[i("a",{staticClass:"header-anchor",attrs:{href:"#citation"}},[e._v("#")]),e._v(" Citation")]),e._v(" "),i("p",[e._v("Are you using TRiCoLOR in your works? Please cite:")]),e._v(" "),i("blockquote",[i("p",[e._v("Davide Bolognini, Alberto Magi, Vladimir Benes, Jan O Korbel, Tobias Rausch. "),i("a",{attrs:{href:"https://academic.oup.com/gigascience/article/9/10/giaa101/5918863?searchresult=1",target:"_blank",rel:"noopener noreferrer"}},[e._v("TRiCoLOR: tandem repeat profiling using whole-genome long-read sequencing data"),i("OutboundLink")],1),e._v(". GigaScience. 2020 Oct 7.")])])])}),[],!1,null,null,null);t.default=a.exports}}]);