<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>General usage | TRiCoLOR documentation</title>
    <meta name="description" content="">
    <meta name="generator" content="VuePress 1.3.1">
    
    
    <link rel="preload" href="/tricolordoc/assets/css/0.styles.aba1ef85.css" as="style"><link rel="preload" href="/tricolordoc/assets/js/app.4d01d9d9.js" as="script"><link rel="preload" href="/tricolordoc/assets/js/3.16cab0cf.js" as="script"><link rel="preload" href="/tricolordoc/assets/js/8.3b10eb58.js" as="script"><link rel="prefetch" href="/tricolordoc/assets/js/10.8db3063e.js"><link rel="prefetch" href="/tricolordoc/assets/js/4.9229c9f2.js"><link rel="prefetch" href="/tricolordoc/assets/js/5.8a7ac43b.js"><link rel="prefetch" href="/tricolordoc/assets/js/6.e9e427aa.js"><link rel="prefetch" href="/tricolordoc/assets/js/7.a8eb3926.js"><link rel="prefetch" href="/tricolordoc/assets/js/9.960c0837.js"><link rel="prefetch" href="/tricolordoc/assets/js/vendors~docsearch.437a26f8.js">
    <link rel="stylesheet" href="/tricolordoc/assets/css/0.styles.aba1ef85.css">
  </head>
  <body>
    <div id="app" data-server-rendered="true"><div class="theme-container"><header class="navbar"><div class="sidebar-button"><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512" class="icon"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z"></path></svg></div> <a href="/tricolordoc/" class="home-link router-link-active"><!----> <span class="site-name">TRiCoLOR documentation</span></a> <div class="links"><form id="search-form" role="search" class="algolia-search-wrapper search-box"><input id="algolia-search-input" class="search-query"></form> <nav class="nav-links can-hide"><div class="nav-item"><a href="/tricolordoc/" class="nav-link">
  Home
</a></div><div class="nav-item"><a href="/tricolordoc/installation/installation.html" class="nav-link">
  Installation
</a></div><div class="nav-item"><a href="/tricolordoc/usage/usage.html" class="nav-link router-link-exact-active router-link-active">
  General usage
</a></div><div class="nav-item"><a href="/tricolordoc/usecases/usecases.html" class="nav-link">
  Use cases
</a></div><div class="nav-item"><a href="https://github.com/davidebolo1993/TRiCoLOR.git" target="_blank" rel="noopener noreferrer" class="nav-link external">
  GitHub
  <svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a></div> <!----></nav></div></header> <div class="sidebar-mask"></div> <aside class="sidebar"><nav class="nav-links"><div class="nav-item"><a href="/tricolordoc/" class="nav-link">
  Home
</a></div><div class="nav-item"><a href="/tricolordoc/installation/installation.html" class="nav-link">
  Installation
</a></div><div class="nav-item"><a href="/tricolordoc/usage/usage.html" class="nav-link router-link-exact-active router-link-active">
  General usage
</a></div><div class="nav-item"><a href="/tricolordoc/usecases/usecases.html" class="nav-link">
  Use cases
</a></div><div class="nav-item"><a href="https://github.com/davidebolo1993/TRiCoLOR.git" target="_blank" rel="noopener noreferrer" class="nav-link external">
  GitHub
  <svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a></div> <!----></nav>  <ul class="sidebar-links"><li><a href="/tricolordoc/installation/installation.html" class="sidebar-link">Installation</a></li><li><a href="/tricolordoc/usage/usage.html" class="active sidebar-link">General usage</a><ul class="sidebar-sub-headers"><li class="sidebar-sub-header"><a href="/tricolordoc/usage/usage.html#tricolor-sensor" class="sidebar-link">TRiCoLOR SENSoR</a></li><li class="sidebar-sub-header"><a href="/tricolordoc/usage/usage.html#tricolor-refer" class="sidebar-link">TRiCoLOR REFER</a></li><li class="sidebar-sub-header"><a href="/tricolordoc/usage/usage.html#tricolor-sage" class="sidebar-link">TRiCoLOR SAGE</a></li><li class="sidebar-sub-header"><a href="/tricolordoc/usage/usage.html#tricolor-app" class="sidebar-link">TRiCoLOR ApP</a></li></ul></li><li><a href="/tricolordoc/usecases/usecases.html" class="sidebar-link">Use cases</a></li></ul> </aside> <main class="page"> <div class="theme-default-content content__default"><h1 id="general-usage"><a href="#general-usage" class="header-anchor">#</a> General usage</h1> <p><a href="https://github.com/davidebolo1993/TRiCoLOR" target="_blank" rel="noopener noreferrer">TRiCoLOR<svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a> is built on 4 modules:</p> <ol><li><a href="https://github.com/davidebolo1993/TRiCoLOR/tree/master/TRiCoLOR/SENSoR" target="_blank" rel="noopener noreferrer">SENSoR<svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a>, the 'Shannon ENtropy ScanneR'</li> <li><a href="https://github.com/davidebolo1993/TRiCoLOR/tree/master/TRiCoLOR/REFER" target="_blank" rel="noopener noreferrer">REFER<svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a>, the 'REpeats FindER'</li> <li><a href="https://github.com/davidebolo1993/TRiCoLOR/tree/master/TRiCoLOR/SAGE" target="_blank" rel="noopener noreferrer">SAGE<svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a>, the 'SAmple GEnotyper'</li> <li><a href="https://github.com/davidebolo1993/TRiCoLOR/tree/master/TRiCoLOR/ApP" target="_blank" rel="noopener noreferrer">ApP<svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a>, the 'Alignment Plotter'</li></ol> <h2 id="tricolor-sensor"><a href="#tricolor-sensor" class="header-anchor">#</a> TRiCoLOR SENSoR</h2> <p>SENSoR allows users to identify repetitive regions from haplotype-resolved and haplotype-tagged <a href="https://samtools.github.io/hts-specs/SAMv1.pdf" target="_blank" rel="noopener noreferrer">BAM files<svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a> de novo (that is, without a prior knowledge of their location). This is achieved using the <a href="https://www.hindawi.com/journals/mpe/2012/132625/" target="_blank" rel="noopener noreferrer">Shannon Entropy formula<svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a>, which allows to discriminate between high-entropy non-repetitive and low-entropy repetitive DNA stretches.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token comment">#TRiCoLOR SENSoR --help</span>

<span class="token comment">#with haplotype-resolved BAM</span>
TRiCoLOR SENSoR -bam <span class="token operator">&lt;</span>HAPLOTYPE1.BAM<span class="token operator">&gt;</span> <span class="token operator">&lt;</span>HAPLOTYPE2.BAM<span class="token operator">&gt;</span> -o <span class="token operator">&lt;</span>OUTPUTSENSOR<span class="token operator">&gt;</span>

<span class="token comment">#with haplotype-tagged BAM</span>
TRiCoLOR SENSoR -bam <span class="token operator">&lt;</span>HAPLOTAGGED.BAM<span class="token operator">&gt;</span> -o <span class="token operator">&lt;</span>OUTPUTSENSOR<span class="token operator">&gt;</span>
</code></pre></div><p>SENSoR stores in the output folder the repetitive regions identified in (gzipped) <a href="https://www.ensembl.org/info/website/upload/bed.html" target="_blank" rel="noopener noreferrer">BED format<svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a>. Specifically, the BED file generated contains 6 columns:</p> <ol><li>Chromosome identifier</li> <li>Start coordinate</li> <li>End coordinate</li> <li>Average coverage. Average of the number of entropy drops found on the two haplotypes</li> <li>Standard deviation of the coverage. Standard deviation of the number of entropy drops found on the two haplotypes</li> <li>Individual coverages. Number of entropy drops found on the two haplotypes</li></ol> <p>Use cases for SENSoR are described in the <a href="/tricolordoc/usecases/usecases.html">Use cases section</a>.</p> <div class="custom-block tip"><p class="custom-block-title">TIP</p> <p>While this is not mandatory, it is highly recommended to give as input to SENSoR also a .txt file with the names of the chromosomes to exclude from analysis (--exclude parameter) Ideally, this should include decoy chromosomes, that harbor low complexity sequences and may be a source of bias for TRiCoLOR</p></div> <h2 id="tricolor-refer"><a href="#tricolor-refer" class="header-anchor">#</a> TRiCoLOR REFER</h2> <p>REFER allows users to profile tandem repeats in haplotype-resolved and haplotype-tagged BAM files once a proper BED file (like the one from TRiCoLOR SENSoR) describing regions to investigate is provided. For each haplotype, REFER fetches sequencing reads spanning regions in the BED file and builds low-error consensus sequences that are further processed through a RegEx-based approximate string matching algorithm to resolve repeated motifs and number of repetitions.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token comment">#TRiCoLOR REFER --help</span>

<span class="token comment">#with haplotype-resolved BAM</span>
TRiCoLOR REFER -g <span class="token operator">&lt;</span>REFERENCE.FASTA<span class="token operator">&gt;</span> -bam <span class="token operator">&lt;</span>HAPLOTYPE1.BAM<span class="token operator">&gt;</span> <span class="token operator">&lt;</span>HAPLOTYPE2.BAM<span class="token operator">&gt;</span> -bed <span class="token operator">&lt;</span>REPETITIONS.BED<span class="token operator">&gt;</span> -o <span class="token operator">&lt;</span>OUTPUTREFER<span class="token operator">&gt;</span>

<span class="token comment">#with haplotype-tagged BAM</span>
TRiCoLOR REFER -g <span class="token operator">&lt;</span>REFERENCE.FASTA<span class="token operator">&gt;</span> -bam <span class="token operator">&lt;</span>HAPLOTAGGED.BAM<span class="token operator">&gt;</span> -bed <span class="token operator">&lt;</span>REPETITIONS.BED<span class="token operator">&gt;</span> -o <span class="token operator">&lt;</span>OUTPUTREFER<span class="token operator">&gt;</span>
</code></pre></div><p>REFER stores in the output folder the tandem repeats varying between the reference and the haplotypes in (bgzipped) <a href="https://samtools.github.io/hts-specs/VCFv4.3.pdf" target="_blank" rel="noopener noreferrer">VCF format<svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a> and (gzipped) BED format, together with consensus alignments for each haplotype in BAM format. INFO and FORMAT fields of the VCF file contain custom informations.</p> <p>The INFO field in the VCF file contains:</p> <ol><li>TREND.  END coordinate of the TR</li> <li>TRLEN. Length of the ALT allele. If multiple alleles are listed in the ALT field, is the length of the shortest ALT allele</li> <li>RAED. Edit distance between REF and ALT. If multiple alleles are listed in the ALT field, is the edit ditance between REF and the most similar ALT allele</li> <li>AED. Edit distance between ALT alleles (if multiple alleles are listed in the ALT field)</li> <li>MAPQ1. Mapping quality of the consensus sequence generated for the 1st haplotype</li> <li>MAPQ2. Mapping quality of the consensus sequence generated for the 2nd haplotype</li> <li>H1M. Repeated motif(s) found on the 1st haplotype</li> <li>H1N. Number of repetitions of the motif(s) in H1M</li> <li>H2M. Repeated motif(s) found on the 2nd haplotype</li> <li>H2N. Number of repetitions of the motif(s) in H2M</li></ol> <p>The FORMAT field contains:</p> <ol><li>GT. The phased genotype of the TR</li> <li>DP1. Depth of coverage for the 1st haplotype</li> <li>DP2. Depth of coverage for the 2nd haplotype</li></ol> <p>The BED files generated contain 5 columns:</p> <ol><li>Chromosome identifier</li> <li>Start coordinate</li> <li>End coordinate</li> <li>Repeated motif</li> <li>Number of repetitions</li></ol> <p>When running TRiCoLOR REFER on multiple samples at the same time, in order to avoid problems with creating/loading .mmi indexes for the same chromosome from multiple processes, one can first create all the .mmi indexes required for the analysis as below:</p> <div class="language-bash extra-class"><pre class="language-bash"><code>TRiCoLOR REFER -g <span class="token operator">&lt;</span>REFERENCE.FASTA<span class="token operator">&gt;</span> -bed <span class="token operator">&lt;</span>REPETITIONS.BED<span class="token operator">&gt;</span> -o <span class="token operator">&lt;</span>OUTPUTINDEXES<span class="token operator">&gt;</span> --index_only
</code></pre></div><p>Then, multiple TRiCoLOR REFER can be run giving the pre-compiled indexes as input:</p> <div class="language-bash extra-class"><pre class="language-bash"><code>TRiCoLOR REFER -g <span class="token operator">&lt;</span>REFERENCE.FASTA<span class="token operator">&gt;</span> -bam <span class="token operator">&lt;</span>HAPLOTAGGED.BAM<span class="token operator">&gt;</span> -bed <span class="token operator">&lt;</span>REPETITIONS.BED<span class="token operator">&gt;</span> -o <span class="token operator">&lt;</span>OUTPUTREFER<span class="token operator">&gt;</span> --mmidir <span class="token operator">&lt;</span>OUTPUTINDEXES<span class="token operator">&gt;</span>
</code></pre></div><p>Use cases for REFER are described in the <a href="/tricolordoc/usecases/usecases.html">Use cases section</a>.</p> <div class="custom-block tip"><p class="custom-block-title">TIP</p> <p>While this is not mandatory, it is highly recommended to give as input to REFER also a .txt file with the names of the chromosomes to exclude from analysis (--exclude parameter) Ideally, this should include decoy chromosomes, that harbor low complexity sequences and may be a source of bias for TRiCoLOR</p></div> <h2 id="tricolor-sage"><a href="#tricolor-sage" class="header-anchor">#</a> TRiCoLOR SAGE</h2> <p>SAGE allows users to check for patterns of mendelian segregation in the tandem repeats identified by TRiCoLOR REFER having trio genome sequencing data available. When a VCF file of the tandem repeats identified in a child by REFER and haplotype-resolved or haplotype-tagged BAM files for both the parents are available, one can run SAGE.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token comment">#TRiCoLOR SAGE --help</span>

<span class="token comment">#with haplotype-resolved BAM</span>
TRiCoLOR SAGE -vcf <span class="token operator">&lt;</span>CHILD.REFER.VCF<span class="token operator">&gt;</span> -bam <span class="token operator">&lt;</span>PARENT1.HAPLOTYPE1.BAM<span class="token operator">&gt;</span>,<span class="token operator">&lt;</span>PARENT1.HAPLOTYPE2.BAM<span class="token operator">&gt;</span> <span class="token operator">&lt;</span>PARENT2.HAPLOTYPE1.BAM<span class="token operator">&gt;</span>,<span class="token operator">&lt;</span>PARENT2.HAPLOTYPE2.BAM<span class="token operator">&gt;</span> -o <span class="token operator">&lt;</span>OUTPUTSAGE<span class="token operator">&gt;</span>

<span class="token comment">#with haplotype-tagged BAM</span>
TRiCoLOR SAGE -vcf <span class="token operator">&lt;</span>CHILD.REFER.VCF<span class="token operator">&gt;</span> -bam <span class="token operator">&lt;</span>PARENT1.HAPLOTAGGED.BAM<span class="token operator">&gt;</span> <span class="token operator">&lt;</span>PARENT2.HAPLOTAGGED.BAM<span class="token operator">&gt;</span> -o <span class="token operator">&lt;</span>OUTPUTSAGE<span class="token operator">&gt;</span>
</code></pre></div><p>SAGE stores in the output folder a (bgzipped) multi-sample (child and both parents) VCF file. INFO and FORMAT fields of the VCF file contain custom informations.</p> <p>The INFO field in the VCF file contains:</p> <ol><li>TREND.  END coordinate of the TR</li> <li>TRLEN. Length of the ALT allele. If multiple alleles are listed in the ALT field, is the length of the shortest ALT allele</li> <li>RAED. Edit distance between REF and ALT. If multiple alleles are listed in the ALT field, is the edit ditance between REF and the most similar ALT allele</li> <li>AED. Edit distance between ALT alleles (if multiple alleles are listed in the ALT field)</li> <li>MISSR. Ratio of missing ('.') genotypes</li> <li>MENDEL. Whether the TR is mendelian consistent (0) or not (1), if --mendel was enabled. '.' otherwhise</li></ol> <p>The FORMAT field contains:</p> <ol><li>GT. The phased genotype of the TR</li> <li>DP1. Depth of coverage for the 1st haplotype</li> <li>DP2. Depth of coverage for the 2nd haplotype</li> <li>GS. Edit distance-based score. Each haplotype has a maximum GS of 1: thus, for a genotype, GS ranges between 0 and 2 (always 2 for the child). For each haplotype, GS is calculated as 1 - the edit distance between the parental sequence and the most similar among the REF sequence and the child ALT sequence(s).</li></ol> <p>Use cases for SAGE are described in the <a href="/tricolordoc/usecases/usecases.html">Use cases section</a>.</p> <h2 id="tricolor-app"><a href="#tricolor-app" class="header-anchor">#</a> TRiCoLOR ApP</h2> <p>ApP allows users to interactively visualize specific tandem repeats identified by the REFER module into their sequence context. It takes as inputs the BED files and the consensus BAM files generated by REFER, together with a region in CHROM:START-END format to visualize.</p> <p>ApP can be run as follows.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token comment">#TRiCoLOR ApP --help</span>
TRiCoLOR ApP -g <span class="token operator">&lt;</span>REFERENCE.FASTA<span class="token operator">&gt;</span> -bam <span class="token operator">&lt;</span>HAPLOTYPE1.CONSENSUS.BAM<span class="token operator">&gt;</span> <span class="token operator">&lt;</span>HAPLOTYPE2.CONSENSUS.BAM<span class="token operator">&gt;</span> -gb <span class="token operator">&lt;</span>REFERENCE.REPETITIONS.BED<span class="token operator">&gt;</span> -h1b <span class="token operator">&lt;</span>HAPLOTYPE1.REPETITIONS.BED<span class="token operator">&gt;</span> -h2b <span class="token operator">&lt;</span>HAPLOTYPE2.REPETITIONS.BED<span class="token operator">&gt;</span> -o <span class="token operator">&lt;</span>OUTPUTAPP<span class="token operator">&gt;</span> <span class="token operator">&lt;</span>REGION<span class="token operator">&gt;</span>
</code></pre></div><p>ApP stores in the output folder a static HTML file that can be opened using the default browser. Users can scroll across and zoom into the sequence of the haplotype-specific consensus BAM files for the chosen region and highlight the tandem repeats identified both in the individual's haplotypes and the reference. Use cases for ApP are described in the <a href="/tricolordoc/usecases/usecases.html">Use cases section</a>.</p></div> <footer class="page-edit"><!----> <!----></footer> <div class="page-nav"><p class="inner"><span class="prev">
      ←
      <a href="/tricolordoc/installation/installation.html" class="prev">
        Installation
      </a></span> <span class="next"><a href="/tricolordoc/usecases/usecases.html">
        Use cases
      </a>
      →
    </span></p></div> </main></div><div class="global-ui"></div></div>
    <script src="/tricolordoc/assets/js/app.4d01d9d9.js" defer></script><script src="/tricolordoc/assets/js/3.16cab0cf.js" defer></script><script src="/tricolordoc/assets/js/8.3b10eb58.js" defer></script>
  </body>
</html>
