<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>Use cases | TRiCoLOR documentation</title>
    <meta name="description" content="">
    <meta name="generator" content="VuePress 1.3.1">
    
    
    <link rel="preload" href="/tricolordoc/assets/css/0.styles.aba1ef85.css" as="style"><link rel="preload" href="/tricolordoc/assets/js/app.4d01d9d9.js" as="script"><link rel="preload" href="/tricolordoc/assets/js/3.16cab0cf.js" as="script"><link rel="preload" href="/tricolordoc/assets/js/9.960c0837.js" as="script"><link rel="prefetch" href="/tricolordoc/assets/js/10.8db3063e.js"><link rel="prefetch" href="/tricolordoc/assets/js/4.9229c9f2.js"><link rel="prefetch" href="/tricolordoc/assets/js/5.8a7ac43b.js"><link rel="prefetch" href="/tricolordoc/assets/js/6.e9e427aa.js"><link rel="prefetch" href="/tricolordoc/assets/js/7.a8eb3926.js"><link rel="prefetch" href="/tricolordoc/assets/js/8.3b10eb58.js"><link rel="prefetch" href="/tricolordoc/assets/js/vendors~docsearch.437a26f8.js">
    <link rel="stylesheet" href="/tricolordoc/assets/css/0.styles.aba1ef85.css">
  </head>
  <body>
    <div id="app" data-server-rendered="true"><div class="theme-container"><header class="navbar"><div class="sidebar-button"><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512" class="icon"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z"></path></svg></div> <a href="/tricolordoc/" class="home-link router-link-active"><!----> <span class="site-name">TRiCoLOR documentation</span></a> <div class="links"><form id="search-form" role="search" class="algolia-search-wrapper search-box"><input id="algolia-search-input" class="search-query"></form> <nav class="nav-links can-hide"><div class="nav-item"><a href="/tricolordoc/" class="nav-link">
  Home
</a></div><div class="nav-item"><a href="/tricolordoc/installation/installation.html" class="nav-link">
  Installation
</a></div><div class="nav-item"><a href="/tricolordoc/usage/usage.html" class="nav-link">
  General usage
</a></div><div class="nav-item"><a href="/tricolordoc/usecases/usecases.html" class="nav-link router-link-exact-active router-link-active">
  Use cases
</a></div><div class="nav-item"><a href="https://github.com/davidebolo1993/TRiCoLOR.git" target="_blank" rel="noopener noreferrer" class="nav-link external">
  GitHub
  <svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a></div> <!----></nav></div></header> <div class="sidebar-mask"></div> <aside class="sidebar"><nav class="nav-links"><div class="nav-item"><a href="/tricolordoc/" class="nav-link">
  Home
</a></div><div class="nav-item"><a href="/tricolordoc/installation/installation.html" class="nav-link">
  Installation
</a></div><div class="nav-item"><a href="/tricolordoc/usage/usage.html" class="nav-link">
  General usage
</a></div><div class="nav-item"><a href="/tricolordoc/usecases/usecases.html" class="nav-link router-link-exact-active router-link-active">
  Use cases
</a></div><div class="nav-item"><a href="https://github.com/davidebolo1993/TRiCoLOR.git" target="_blank" rel="noopener noreferrer" class="nav-link external">
  GitHub
  <svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a></div> <!----></nav>  <ul class="sidebar-links"><li><a href="/tricolordoc/installation/installation.html" class="sidebar-link">Installation</a></li><li><a href="/tricolordoc/usage/usage.html" class="sidebar-link">General usage</a></li><li><a href="/tricolordoc/usecases/usecases.html" class="active sidebar-link">Use cases</a><ul class="sidebar-sub-headers"><li class="sidebar-sub-header"><a href="/tricolordoc/usecases/usecases.html#tricolor-sensor" class="sidebar-link">TRiCoLOR SENSoR</a></li><li class="sidebar-sub-header"><a href="/tricolordoc/usecases/usecases.html#tricolor-refer" class="sidebar-link">TRiCoLOR REFER</a></li><li class="sidebar-sub-header"><a href="/tricolordoc/usecases/usecases.html#tricolor-app" class="sidebar-link">TRiCoLOR ApP</a></li><li class="sidebar-sub-header"><a href="/tricolordoc/usecases/usecases.html#tricolor-sage" class="sidebar-link">TRiCoLOR SAGE</a></li></ul></li></ul> </aside> <main class="page"> <div class="theme-default-content content__default"><h1 id="use-cases"><a href="#use-cases" class="header-anchor">#</a> Use cases</h1> <p>Following are described some use cases, examples illustrating how to run TRiCoLOR modules. To this purpose, a test set is provided in the TRiCoLOR/test folder. The folder contains BAM files simulated with <a href="https://academic.oup.com/bioinformatics/article/doi/10.1093/bioinformatics/btz719/5582674/" target="_blank" rel="noopener noreferrer">VISOR<svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a> (accuracy ~ 0.9, reads length ~ 8000 bps, substitutions:insertions:deletions rate ~ 45:25:30, coverage ~ 50X) for a trio (SON, PARENT1 and PARENT2). BAM files for SON and PARENT1 contain a heterozigous TR expansion. In particular, the normal TR is chr20:17553794-17553824 (15xAC), taken from the list of TRs availables for the GRCh38 human reference genome, whereas the expanded TR is 115XAC (expansion of 100 motifs).</p> <h2 id="tricolor-sensor"><a href="#tricolor-sensor" class="header-anchor">#</a> TRiCoLOR SENSoR</h2> <p>We can identify repetitive regions in SON using the SENSoR module. As described in the <a href="/tricolordoc/usage/usage.html">General usage section</a>, SENSoR requires haplotype-resolved or haplotype-tagged BAM files. For instance, using the haplotype-tagged BAM generated by VISOR, SENSoR can be run as follows:</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token comment">#from TRiCoLOR/test</span>
TRiCoLOR SENSoR -bam son/sim.srt.bam -o sensor_son
</code></pre></div><p>This will produce a gzipped BED file in the output folder, containing repetitive regions identified in the initial haplotype-resolved BAM files. As we are interested on further profiling a single TR, which is known to occur at chr20:17553794-17553824, we can ignore other regions.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token comment">#from TRiCoLOR/test</span>
zgrep -P <span class="token string">&quot;^chr20<span class="token entity" title="\t">\t</span>17553&quot;</span> sensor_son/TRiCoLOR.srt.bed.gz <span class="token operator">&gt;</span> sensor_son/TRiCoLOR.srt.fltrd.bed
</code></pre></div><h2 id="tricolor-refer"><a href="#tricolor-refer" class="header-anchor">#</a> TRiCoLOR REFER</h2> <p>We can profile the TR included in the BED file previously generated using the REFER module. As described in the <a href="/tricolordoc/usage/usage.html">General usage section</a>, in addition to the haplotype-tagged BAM file and the BED file we already generated, REFER requires a reference genome. The one used to simulate the BAM files included in in the TRiCoLOR/test folder is the GRCh38 human reference genome.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token comment">#from TRiCoLOR/test</span>
<span class="token function">wget</span> ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/GRCh38_reference_genome/GRCh38_full_analysis_set_plus_decoy_hla.fa
samtools faidx GRCh38_full_analysis_set_plus_decoy_hla.fa
</code></pre></div><p>Then, REFER can be run as follows. As the normal TR size is 30 bp and by default TRiCoLOR only outputs TRs larger than 50 bp, we lower this treshold by setting the -s/--size parameter to 20.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token comment">#from TRiCoLOR/test</span>
TRiCoLOR REFER -g GRCh38_full_analysis_set_plus_decoy_hla.fa -bam son/sim.srt.bam -bed sensor_son/TRiCoLOR.srt.fltrd.bed -o refer_son -s <span class="token number">20</span> --samplename SON
</code></pre></div><p>This will produce a bgzipped VCF file containing the variant TR.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token comment">#from TRiCoLOR/test</span>
bcftools view refer_son/TRiCoLOR.srt.vcf.gz
</code></pre></div><p>As can be seen from the VCF file, the first haplotype contains a normal TR (H1M:AC; H1N:15) whereas the second haplotype an expanded TR (H2M:AC; H2N:117), which almost perfectly approximate the ground truth (H1M:AC; H1N:15; H2M:AC; H2N:115). Together with the VCF file, TRiCoLOR outputs in BED format the TRs found in the reference (sensor_refer/reference/TRiCoLOR.srt.bed.gz) and in the individuals' haplotypes (sensor_refer/haplotype1/TRiCoLOR.srt.bed.gz, sensor_refer/haplotype2/TRiCoLOR.srt.bed.gz) as well as the haplotype-resolved consensus alignments (sensor_refer/haplotype1/TRiCoLOR.srt.bam, sensor_refer/haplotype2/TRiCoLOR.srt.bam). These additional files can be used to interactively visualize the TRs identified into their sequence context.</p> <h2 id="tricolor-app"><a href="#tricolor-app" class="header-anchor">#</a> TRiCoLOR ApP</h2> <p>We can interactively visualize the TRs identified using the ApP module. As described in the <a href="/tricolordoc/usage/usage.html">General usage section</a>, ApP requires, together with the reference genome, the BED and the consensus BAM files generated by the REFER module, a region specifying a TR of interest.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token comment">#from TRiCoLOR/test</span>
TRiCoLOR ApP -g GRCh38_full_analysis_set_plus_decoy_hla.fa -bam refer_son/haplotype1/TRiCoLOR.srt.bam refer_son/haplotype2/TRiCoLOR.srt.bam -o app_son -gb refer_son/reference/TRiCoLOR.srt.bed.gz -h1b refer_son/haplotype1/TRiCoLOR.srt.bed.gz -h2b refer_son/haplotype2/TRiCoLOR.srt.bed.gz chr20:17553795-17553825
</code></pre></div><p>This will produce an HTML file in the output folder, that can be opened using the default web browser.</p> <h2 id="tricolor-sage"><a href="#tricolor-sage" class="header-anchor">#</a> TRiCoLOR SAGE</h2> <p>As described in the <a href="/tricolordoc/usage/usage.html">General usage section</a>, TRiCoLOR also provides users with the SAGE module, which checks the mendelian consistency of the TRs identified in a child if haplotype-resolved or haplotype-tagged alignments for both the parents are available. SAGE can be run as follows.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token comment">#from TRiCoLOR/test</span>
TRiCoLOR SAGE -vcf refer_son/TRiCoLOR.srt.vcf.gz -bam parent1/sim.srt.bam parent2/sim.srt.bam -o sage_trio --samplename PARENT1 PARENT2 --mendel
</code></pre></div><p>This will produce a bgzipped multi-sample VCF file containing with parents genotyped.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token comment">#from TRiCoLOR/test</span>
bcftools view sage_trio/TRiCoLOR.srt.vcf.gz
</code></pre></div><p>As can be seen from the VCF file, PARENT1 contains an heterozigous TR expansions while PARENT2 is homozigous reference. The TR identified in the child is threrefore considered mendelian consistent (MENDEL:0).</p></div> <footer class="page-edit"><!----> <!----></footer> <div class="page-nav"><p class="inner"><span class="prev">
      ←
      <a href="/tricolordoc/usage/usage.html" class="prev">
        General usage
      </a></span> <!----></p></div> </main></div><div class="global-ui"></div></div>
    <script src="/tricolordoc/assets/js/app.4d01d9d9.js" defer></script><script src="/tricolordoc/assets/js/3.16cab0cf.js" defer></script><script src="/tricolordoc/assets/js/9.960c0837.js" defer></script>
  </body>
</html>
